packages/core/admin/admin/src/content-manager/pages/ListView/index.js

```diff
@@ -180,6 +180,22 @@ function ListView({
           data: { results, pagination: paginationResult },
         } = await fetchClient.get(endPoint, options);
 
+        // If user enters a page number that doesn't exist, redirect him to the last page
+        if (paginationResult.page > paginationResult.pageCount && paginationResult.pageCount > 0) {
+          const query = {
+            ...params,
+            page: paginationResult.pageCount,
+          };
+
+          push({
+            pathname,
+            state: { from: pathname },
+            search: stringify(query),
```

Comment:
Please disable encoding here, its updating url.
```suggestion
            search: stringify(query, { encode: false }),
```